allensdk/brain_observatory/behavior/data_objects/projections.py

```diff
@@ -0,0 +1,154 @@
+        return Projections(max_projection=max_projection,
+                           avg_projection=avg_projection)
+
+    @classmethod
+    def from_nwb(cls, nwbfile: NWBFile) -> "Projections":
+        max_projection = get_image(nwbfile=nwbfile, name='max_projection',
+                                   module='ophys')
+        avg_projection = get_image(nwbfile=nwbfile, name='average_image',
+                                   module='ophys')
+        return Projections(max_projection=max_projection,
+                           avg_projection=avg_projection)
+
+    def to_nwb(self, nwbfile: NWBFile) -> NWBFile:
+        def _add_image(image_data: Image, image_name: str):
+            MODULE_NAME = 'ophys'
+            description = '{} image at pixels/cm resolution'.format(image_name)
+
+            data, spacing, unit = image_data
+
+            assert spacing[0] == spacing[1] and len(
+                spacing) == 2 and unit == 'mm'
+
+            if MODULE_NAME not in nwbfile.processing:
+                ophys_mod = ProcessingModule(MODULE_NAME,
+                                             'Ophys processing module')
+                nwbfile.add_processing_module(ophys_mod)
+            else:
+                ophys_mod = nwbfile.processing[MODULE_NAME]
+
+            image = GrayscaleImage(image_name,
+                                   data,
+                                   resolution=spacing[0] / 10,
+                                   description=description)
+
+            if 'images' not in ophys_mod.containers:
+                images = Images(name='images')
+                ophys_mod.add_data_interface(images)
+            else:
+                images = ophys_mod['images']
+            images.add_image(image)
+
+        _add_image(image_data=self._max_projection,
+                   image_name='max_projection')
+        _add_image(image_data=self._avg_projection, image_name='average_image')
+
+        return nwbfile
+
+    @classmethod
+    def from_json(cls, dict_repr: dict) -> "Projections":
+        max_projection_filepath = dict_repr['max_projection_file']
+        avg_projection_filepath = \
+            dict_repr['average_intensity_projection_image_file']
+        pixel_size = dict_repr['surface_2p_pixel_size_um']
+
+        max_projection = cls._from_filepath(filepath=max_projection_filepath,
+                                            pixel_size=pixel_size)
+        avg_projection = cls._from_filepath(filepath=avg_projection_filepath,
+                                            pixel_size=pixel_size)
+        return Projections(max_projection=max_projection,
+                           avg_projection=avg_projection)
+
+    @staticmethod
+    def _from_filepath(filepath: str, pixel_size: float) -> Image:
+        """
+        :param filepath
+            path to image
+        :param pixel_size
+            pixel size in um
+        """
+        max_projection = mpimg.imread(filepath)
```

Comment:
`max_projection` name is too specific here, as this method can be used for either type of image, I think

Reply:
good catch, corrected